jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java

```diff
@@ -82,28 +83,62 @@ public SecureRequestCustomizer(
         @Name("stsMaxAgeSeconds") long stsMaxAgeSeconds,
         @Name("stsIncludeSubdomains") boolean stsIncludeSubdomains)
     {
+        this(false, sniHostCheck, stsMaxAgeSeconds, stsIncludeSubdomains);
+    }
+
+    /**
+     * @param sniRequired True if a SNI certificate is required.
+     * @param sniHostCheck True if the SNI Host name must match.
+     * @param stsMaxAgeSeconds The max age in seconds for a Strict-Transport-Security response header. If set less than zero then no header is sent.
+     * @param stsIncludeSubdomains If true, a include subdomain property is sent with any Strict-Transport-Security header
+     */
+    public SecureRequestCustomizer(
+        @Name("sniRequired") boolean sniRequired,
+        @Name("sniHostCheck") boolean sniHostCheck,
+        @Name("stsMaxAgeSeconds") long stsMaxAgeSeconds,
+        @Name("stsIncludeSubdomains") boolean stsIncludeSubdomains)
+    {
+        _sniRequired = sniRequired;
```

Comment:
If an arg is passed in to a constructor generally speaking means that it is not expected to be modified by a setter, but yet there is setSNIRequired method ....

Reply:
I  kind of agree that it should be constructor or setter injection, but that was the style that was.